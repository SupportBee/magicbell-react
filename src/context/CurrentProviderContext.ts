import { createContext } from 'react';

export const NO_PROVIDER = 'MAGICBELL_NO_PROVIDER';

const CurrentProviderContext = createContext<String>(NO_PROVIDER);
export default CurrentProviderContext;
