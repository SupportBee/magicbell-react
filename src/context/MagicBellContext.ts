import { createContext, useContext } from 'react';

export interface IMagicBellContext {
  images?: Partial<{
    emptyInboxUrl: string;
  }>;
}

const MagicBellContext = createContext<IMagicBellContext>({} as IMagicBellContext);
export default MagicBellContext;

const useMagicBellContext = () => useContext(MagicBellContext);
export { useMagicBellContext };
