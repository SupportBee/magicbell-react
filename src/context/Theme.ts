export interface IElementTheme {
  backgroundColor: string;
  backgroundOpacity: number;
  borderRadius: string;
  fontFamily: string;
  fontSize: string;
  textAlign: 'left' | 'right' | 'center' | 'justify' | 'initial' | 'inherit';
  textColor: string;
  textTransform: 'none' | 'capitalize' | 'uppercase' | 'lowercase' | 'initial' | 'inherit';
}

export interface IMagicBellTheme {
  icon: {
    borderColor: string;
    width: string;
  };
  unseenBadge: IElementTheme;
  header: IElementTheme;
  container: IElementTheme;
  footer: IElementTheme;
  notification: {
    default: IElementTheme;
    unread: IElementTheme;
    unseen: IElementTheme;
  };
}

const blue = '#3498F4';
const red = '#DF4759';
const black = '#3A424D';

// prettier-ignore
const fontFamily = '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif';
const fontSize = '14px';
const textAlign = 'left';
const textTransform = 'none';

export const defaultTheme = {
  icon: {
    borderColor: blue,
    width: '24px',
  },
  header: {
    backgroundColor: blue,
    backgroundOpacity: 1,
    borderRadius: '8px',
    fontFamily,
    fontSize,
    textAlign,
    textColor: 'white',
    textTransform,
  },
  footer: {
    backgroundColor: blue,
    backgroundOpacity: 1,
    borderRadius: '8px',
    fontFamily,
    fontSize: fontSize,
    textAlign,
    textColor: 'white',
    textTransform,
  },
  unseenBadge: {
    backgroundColor: red,
    backgroundOpacity: 1,
    borderRadius: '2px',
    fontFamily,
    fontSize: '10px',
    textAlign: 'center',
    textColor: 'white',
    textTransform,
  },
  container: {
    backgroundColor: '#FFFFFF',
    backgroundOpacity: 1,
    borderRadius: '8px',
    fontFamily,
    fontSize: '14px',
    textAlign,
    textColor: black,
    textTransform,
  },
  notification: {
    default: {
      backgroundColor: blue,
      backgroundOpacity: 0.1,
      borderRadius: '8px',
      fontFamily,
      fontSize,
      textAlign,
      textColor: black,
      textTransform,
    },
    unread: {
      backgroundColor: blue,
      backgroundOpacity: 0,
      borderRadius: '8px',
      fontFamily,
      fontSize,
      textAlign,
      textColor: black,
      textTransform,
    },
    unseen: {
      backgroundColor: blue,
      backgroundOpacity: 0.05,
      borderRadius: '8px',
      fontFamily,
      fontSize,
      textAlign,
      textColor: black,
      textTransform,
    },
  },
} as IMagicBellTheme;
