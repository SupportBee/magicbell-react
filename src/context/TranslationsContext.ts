import { createContext, useContext } from 'react';

const TranslationsContext = createContext<any>({});
export default TranslationsContext;

const TranslationsProvider = TranslationsContext.Provider;
const useTranslations = () => useContext(TranslationsContext);
export { TranslationsProvider, useTranslations };
