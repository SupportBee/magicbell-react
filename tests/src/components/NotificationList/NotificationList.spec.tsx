import { act, render, RenderResult } from '@testing-library/react';
import React from 'react';
import Sinon from 'sinon';
import NotificationList, { ListItemProps } from '../../../../src/components/NotificationList';
import { MagicBellThemeProvider } from '../../../../src/context/MagicBellThemeContext';
import { defaultTheme } from '../../../../src/context/Theme';
import { sampleNotification } from '../../../factories/NotificationFactory';

describe('components', () => {
  describe('NotificationList', () => {
    let notifications;
    let onItemClick: (notification: Notification) => void;
    let view: RenderResult;

    beforeEach(() => {
      notifications = {};
      onItemClick = jest.fn();

      view = render(
        <MagicBellThemeProvider value={defaultTheme}>
          <NotificationList notifications={notifications} onItemClick={onItemClick} />
        </MagicBellThemeProvider>,
      );
    });

    afterEach(() => {
      view.unmount();
    });

    describe('render', () => {
      describe('the collection is empty', () => {
        it('renders an empty list', () => {
          expect(view.container).toMatchSnapshot();
        });
      });
    });

    describe('the collection is not empty', () => {
      let clock;

      beforeEach(async () => {
        // The relative time has to be deterministic for snapshots.
        clock = Sinon.useFakeTimers();

        await act(async () => {
          notifications.push(sampleNotification);
        });
      });

      afterEach(() => {
        clock.restore();
      });

      it('renders a virtual list with all items', () => {
        expect(view.container).toMatchSnapshot();
      });
    });

    describe('a height is provided', () => {
      it('renders a list with the given height', () => {
        view.rerender(
          <MagicBellThemeProvider value={defaultTheme}>
            <NotificationList
              notifications={notifications}
              onItemClick={onItemClick}
              height={300}
            />
          </MagicBellThemeProvider>,
        );

        expect(view.container).toMatchSnapshot();
      });
    });

    describe('a custom ListItem is provided', () => {
      beforeEach(async () => {
        await act(async () => {
          notifications.push(sampleNotification);
        });
      });

      it('renders a list rendering the custom component for each item', () => {
        function CustomListItem(props: ListItemProps) {
          return <p>{props.notification.title}</p>;
        }

        view.rerender(
          <MagicBellThemeProvider value={defaultTheme}>
            <NotificationList
              notifications={notifications}
              onItemClick={onItemClick}
              height={300}
              ListItem={CustomListItem}
            />
          </MagicBellThemeProvider>,
        );

        expect(view.container).toMatchSnapshot();
      });
    });
  });
});
