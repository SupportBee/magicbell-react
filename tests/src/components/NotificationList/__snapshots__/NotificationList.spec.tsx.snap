// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components NotificationList a custom ListItem is provided renders a list rendering the custom component for each item 1`] = `
<div>
  <div
    class="infinite-scroll-component__outerdiv"
  >
    <div
      class="infinite-scroll-component "
      style="height: 300px; overflow: auto;"
    >
      <p>
        New Comment: Tables in knowledgebase
      </p>
    </div>
  </div>
</div>
`;

exports[`components NotificationList a height is provided renders a list with the given height 1`] = `
<div>
  <div
    class="infinite-scroll-component__outerdiv"
  >
    <div
      class="infinite-scroll-component "
      style="height: 300px; overflow: auto;"
    />
  </div>
</div>
`;

exports[`components NotificationList render the collection is empty renders an empty list 1`] = `
<div>
  <div
    class="infinite-scroll-component__outerdiv"
  >
    <div
      class="infinite-scroll-component "
      style="height: auto; overflow: auto;"
    />
  </div>
</div>
`;

exports[`components NotificationList the collection is not empty renders a virtual list with all items 1`] = `
.emotion-4 {
  -webkit-backface-visibility: visible;
  backface-visibility: visible;
  background-clip: border-box;
  background-color: transparent;
  background-origin: padding-box;
  background-position: 0 0;
  background-size: auto;
  border-radius: 0;
  border: 0;
  box-shadow: none;
  box-sizing: content-box;
  clear: none;
  -webkit-clip: auto;
  clip: auto;
  color: inherit;
  cursor: auto;
  direction: inherit;
  display: block;
  left: auto;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen-Sans,Ubuntu,Cantarell,'Helvetica Neue', sans-serif;
  font-size: 14px;
  font-weight: normal;
  height: auto;
  -webkit-hyphens: manual;
  -moz-hyphens: manual;
  -ms-hyphens: manual;
  hyphens: manual;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1;
  margin: 0;
  max-height: none;
  max-width: none;
  min-height: 0;
  min-width: 0;
  opacity: 1;
  outline: invert none medium;
  padding: 0;
  -webkit-perspective: none;
  -moz-perspective: none;
  -ms-perspective: none;
  perspective: none;
  position: static;
  right: auto;
  top: auto;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  visibility: inherit;
  white-space: normal;
  width: auto;
  word-spacing: normal;
  z-index: auto;
  background-color: rgba(52,152,244,0) !important;
  border-radius: 8px;
  color: #3A424D !important;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif !important;
  font-size: 14px;
  margin: 4px !important;
  overflow: hidden;
  text-align: left !important;
  text-transform: none;
  -webkit-transition: background-color 300ms ease-out;
  transition: background-color 300ms ease-out;
  padding: 16px 20px !important;
  min-height: 32px;
}

.emotion-4 div,
.emotion-4 span,
.emotion-4 p,
.emotion-4 a,
.emotion-4 strong,
.emotion-4 b,
.emotion-4 u,
.emotion-4 i,
.emotion-4 ol,
.emotion-4 ul,
.emotion-4 li,
.emotion-4 form,
.emotion-4 label,
.emotion-4 table,
.emotion-4 tr,
.emotion-4 th,
.emotion-4 td,
.emotion-4 article,
.emotion-4 aside,
.emotion-4 footer,
.emotion-4 header,
.emotion-4 nav,
.emotion-4 section {
  margin: 0;
  padding: 0;
  border: 0;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

.emotion-4 a {
  cursor: pointer !important;
  color: inherit !important;
}

.emotion-4 button {
  background: transparent;
  border: none;
  font-size: inherit !important;
  color: inherit;
  padding: 0;
  cursor: pointer !important;
}

.emotion-4:hover {
  background-color: rgba(49,144,232,0.05) !important;
}

.emotion-4 > button {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 0 0 -16px;
  text-align: inherit !important;
}

.emotion-4 > button > div {
  margin: 0 0 0 16px;
}

.emotion-4 > button > div:first-of-type {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-0 {
  cursor: inherit;
  font-weight: 500 !important;
  line-height: 1.2 !important;
  word-break: break-word !important;
}

.emotion-2 {
  padding: 0.5em 0 0.5em 0.25em !important;
}

.emotion-3 {
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-align-items: center !important;
  -webkit-box-align: center !important;
  -ms-flex-align: center !important;
  align-items: center !important;
  padding: 0.5em 0 0.5em 0.25em !important;
  color: #3498F4 !important;
}

.emotion-3 svg {
  height: 1.5em;
  fill: currentColor;
}

.emotion-1 {
  color: inherit !important;
  cursor: inherit;
  margin: 0;
  line-height: 1.2;
  font-size: 0.865em !important;
  margin-top: 1em !important;
  word-break: break-word !important;
}

<div>
  <div
    class="infinite-scroll-component__outerdiv"
  >
    <div
      class="infinite-scroll-component "
      style="height: auto; overflow: auto;"
    >
      <div
        class="emotion-4"
      >
        <button
          data-testid="clickable-notification"
        >
          <div>
            <p
              class="emotion-0"
              title="New Comment: Tables in knowledgebase"
            >
              New Comment: Tables in knowledgebase
            </p>
            <article
              class="emotion-1"
            >
              <p>
                This is a good content
              </p>
            </article>
          </div>
          <div
            class="emotion-2"
          >
            <div
              aria-expanded="false"
            >
              50y
            </div>
          </div>
        </button>
        <div
          aria-expanded="false"
          class="emotion-3"
          data-testid="icon"
        >
          <svg
            viewBox="0 0 512 512"
          >
            <circle
              cx="256"
              cy="256"
              r="64"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
